### PR TITLE
[DYOD] Smart Position List: Finalize smart PosList

### PR DESCRIPTION
**Pull request by leogeier**
Renames the RowIdPosList file, renames minor variables, and adds the SmartPosList team to the contributors. These changes were originally part of #2026.

### CODE REVIEW

**Comment by mrks** [comment]
I have a talent of breaking PRs from other repos when I do stuff locally. Could you merge the master, please?

**Comment by mrks** [comment]
(In case there are any conflicts, I'll be happy to have a look)

File: src/lib/resolve_type.hpp [hunk 1]
```diff
@@ -134,8 +134,8 @@ void resolve_pos_list_type(const std::shared_ptr<const AbstractPosList>& untyped
     return;
   }
 
-  if (auto rowid_pos_list = std::dynamic_pointer_cast<const RowIDPosList>(untyped_pos_list)) {
-    func(rowid_pos_list);
+  if (auto row_id_pos_list = std::dynamic_pointer_cast<const RowIDPosList>(untyped_pos_list)) {
```
Comment (Bouncner) [inline src/lib/resolve_type.hpp:137]: Can't that be const?

**Comment by mrks** [comment]
Realized that I could just check out your repo and merge Hyrise there. That feels safer.

**Comment by mrks** [comment]
replaced by #2090

**Pull request closed without merging**